src/hardware/ps1audio.cpp

```diff
@@ -382,22 +384,49 @@ class Ps1Synth {
 	~Ps1Synth();
 
 private:
-	void Update(uint16_t samples);
+	// Block alternate construction routes
+	Ps1Synth()                            = delete;
+	Ps1Synth(const Ps1Synth &)            = delete;
+	Ps1Synth &operator=(const Ps1Synth &) = delete;
+
+	void AudioCallback(uint16_t requested_frames);
+	bool MaybeRenderFrame(float &frame);
+	void RenderUpToNow();
+	int TallySilence(const int sample);
+
 	void WriteSoundGeneratorPort205(io_port_t port, io_val_t, io_width_t);
 
-	mixer_channel_t channel = nullptr;
+	// Managed objects
+	mixer_channel_t channel            = nullptr;
 	IO_WriteHandleObject write_handler = {};
-	static constexpr auto clock_rate_hz = 4000000;
 	sn76496_device device;
-	static constexpr auto max_samples_expected = 64;
-	int16_t buffer[1][max_samples_expected];
-	size_t last_write = 0;
+	std::unique_ptr<reSIDfp::TwoPassSincResampler> resampler = {};
+	std::queue<float> fifo                                   = {};
+	//	static constexpr auto max_samples_expected = 64;
+
+	// Static rate-related configuration
+	static constexpr auto ps1_psg_clock_hz = 4000000;
+	static constexpr auto render_divisor   = 16;
+	static constexpr auto render_rate_hz   = ceil_sdivide(ps1_psg_clock_hz,
+                                                            render_divisor);
+	static constexpr auto ms_per_render    = 1000.0 / render_rate_hz;
+	static constexpr auto idle_after_ms    = 2000;
+
+	// Runtime states
+	device_sound_interface *dsi = static_cast<sn76496_base_device *>(&device);
+	double last_rendered_ms     = 0.0;
+	int idle_after_silent_samples = 0;
+	int silent_samples            = 0;
+	int unused_for_ms             = 0;
+	// bool is_enabled                   = false;
+	/* 	int16_t buffer[1][max_samples_expected];
+	        size_t last_write = 0; */
```

Comment:
Delete these?

Reply:
Good catch, more stuff that fell away with the refactoring!